ThreadJob/ThreadJob/PSThreadJob.cs

```diff
@@ -113,27 +115,188 @@ protected override void EndProcessing()
         #endregion
     }
 
+    public sealed class ThreadJobSourceAdapter : JobSourceAdapter
+    {
+        #region Members
+
+        private ConcurrentDictionary<Guid, Job2> _repository;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public ThreadJobSourceAdapter()
+        {
+            Name = "ThreadJobSourceAdapter";
+            _repository = new ConcurrentDictionary<Guid, Job2>();
+        }
+
+        #endregion
+
+        #region JobSourceAdapter Implementation
+
+        /// <summary>
+        /// NewJob
+        /// </summary>
+        public override Job2 NewJob(JobInvocationInfo specification)
+        {
+            var job = specification.Parameters[0][0].Value as ThreadJob;
+            if (job != null)
+            {
+                _repository.TryAdd(job.InstanceId, job);

```

Comment:
Do we need to worry if TryAdd() fails (returns false)?

Reply:
Job instanceId is a GUID so there will be no key collisions.  I think we can safely ignore.